api/src/test/java/org/openmrs/api/impl/PatientServiceImplUnitTest.java

```diff
@@ -0,0 +1,1379 @@
+
+		verify(administrationService).getGlobalProperty("concept.reasonExitedCare");
+		verify(conceptService).getConcept(property);
+		verify(obsService).getObservationsByPersonAndConcept(patient, concept);
+		verify(locationService).getDefaultLocation();
+		verify(concept, times(1)).getName();
+		verify(obsService).saveObs(any(Obs.class), eq("updated by PatientService.saveReasonForExit"));
+	}
+
+	@Test
+	public void exitFromCare_whenHasNoObservationAndNoDefaultLocation_shouldCreateAnObservationWithoutLocationAndSaveIt() {
+
+		when(administrationService.getGlobalProperty("concept.reasonExitedCare")).thenReturn(property);
+		when(conceptService.getConcept(property)).thenReturn(concept);
+		when(obsService.getObservationsByPersonAndConcept(patient, concept)).thenReturn(observations);
+		when(obsService.saveObs(observation1, "updating")).thenReturn(observation1);
+		when(locationService.getDefaultLocation()).thenReturn(null);
+		when(concept.getName()).thenReturn(conceptName);
+		when(obsService.getObservationsByPersonAndConcept(patient, concept)).thenReturn(new ArrayList<>());
+		patientServiceImpl.exitFromCare(patient, newDate, concept);
+
+		verify(administrationService).getGlobalProperty("concept.reasonExitedCare");
+		verify(conceptService).getConcept(property);
+		verify(obsService).getObservationsByPersonAndConcept(patient, concept);
+		verify(locationService).getDefaultLocation();
+		verify(concept, times(1)).getName();
+		verify(obsService).saveObs(any(Obs.class), eq("updated by PatientService.saveReasonForExit"));
+	}
+
+	@Test
+	public void exitFromCare_whenHasNoReasonFoExit_shouldNotDoOrSaveAnything() {
+
+		when(administrationService.getGlobalProperty("concept.reasonExitedCare")).thenReturn(property);
+		when(conceptService.getConcept(property)).thenReturn(null);
+
+		patientServiceImpl.exitFromCare(patient, newDate, concept);
+
+		verify(administrationService).getGlobalProperty("concept.reasonExitedCare");
+		verify(conceptService).getConcept(property);
+		verify(obsService, never()).getObservationsByPersonAndConcept(patient, concept);
+		verify(locationService, never()).getDefaultLocation();
+		verify(concept, never()).getName();
+		verify(obsService, never()).saveObs(any(Obs.class), eq("updated by PatientService.saveReasonForExit"));
+	}
+
+	//	@Test
+//	public void getPatientOrPromotePerson_whenAPersonIsFoundInsteadOfPatient_shouldPromoteThePersonToPatientAndReturnIt() {
+//		int personId = 1;
+//		Person person = new Person(1);
+//		
+//		when(patientService.getPatient(personId)).thenReturn(person);
+//		when(personService.getPerson(personId)).thenReturn(person);
+//
+//		Patient patient = patientServiceImpl.getPatientOrPromotePerson(personId);
+//		
+//		assertNotNull(patient);
+//		verify(personService).getPerson(personId);
+//	}
+	@Test
+	public void exitFromCare_shouldThrowApiExceptionIfPatientIsNull() {
+
+		expectedException.expect(APIException.class);
+		Patient nullPatient = null;
```

Comment:
Please inline this variable, the name does not give us additional information. the IDE shows us that the first parameter is a Patient. Please do the same for all the tests that are similar to this one.